### PR TITLE
feat(lint): created local rule to use isNullish and nonNullish

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

We would like to enforce the usage of functions `isNullish` and `nonNullish` in the code whenever possible.

# Changes

- Created custom local lint rule for `isNullish` usage.
- Created custom local lint rule for `nonNullish` usage.
- Fix issues created by new rules.

# Tests

It raised issues, correctly, and we fixed it.

### CODE REVIEW

File: src/frontend/src/eth/derived/erc20.derived.ts [hunk 1]
```diff
@@ -116,7 +117,7 @@ export const enabledErc20TokensAddresses: Readable<ContractAddressText[]> = deri
 
 export const erc20UserTokensInitialized: Readable<boolean> = derived(
 	[erc20UserTokensStore],
-	([$erc20UserTokensStore]) => $erc20UserTokensStore !== undefined
+	([$erc20UserTokensStore]) => nonNullish($erc20UserTokensStore)
```
Comment (peterpeterparker) [inline src/frontend/src/eth/derived/erc20.derived.ts:120]: Are you sure this was not checked for `undefined` on purpose given that `null` in store is often used to express the fact that something was explicitely set to `null`?

File: src/frontend/src/icp-eth/services/cketh.services.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ export const loadCkEthMinterInfo = async ({
 	const minterInfoInStore = get(ckEthMinterInfoStore);
 
 	// We try to load only once per session the helpers (ckETH and ckErc20) contract addresses
-	if (minterInfoInStore?.[tokenId] !== undefined) {
+	if (nonNullish(minterInfoInStore?.[tokenId])) {
```
Comment (peterpeterparker) [inline src/frontend/src/icp-eth/services/cketh.services.ts:27]: ditto

File: src/frontend/src/lib/components/ui/Json.svelte [hunk 3]
```diff
@@ -21,7 +22,7 @@
 		| 'undefined';
 
 	const getValueType = (value: unknown): ValueType => {
-		if (value === null) {
+		if (isNullish(value)) {
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/ui/Json.svelte:25]: I think we want to display `undefined` here and the `=== null` is a choice

File: src/frontend/src/lib/derived/address.derived.ts [hunk 4]
```diff
@@ -8,7 +8,7 @@ export const addressNotLoaded: Readable<boolean> = derived([addressStore], ([$ad
 );
 
 export const address: Readable<OptionAddress> = derived([addressStore], ([$addressStore]) =>
-	$addressStore === null ? null : $addressStore?.address
+	isNullish($addressStore) ? null : $addressStore?.address
```
Comment (peterpeterparker) [inline src/frontend/src/lib/derived/address.derived.ts:11]: ditto

File: src/frontend/src/lib/utils/nav.utils.ts [hunk 5]
```diff
@@ -71,7 +71,7 @@ export const loadRouteParams = ($event: LoadEvent): RouteParams => {
 	const token = searchParams?.get('token');
 
 	const replaceEmoji = (input: string | null): string | null => {
-		if (input === null) {
+		if (isNullish(input)) {
```
Comment (peterpeterparker) [inline src/frontend/src/lib/utils/nav.utils.ts:74]: not a call to action, just saying if interesting: input cannot be undefined here

File: src/frontend/src/routes/+layout.svelte [hunk 6]
```diff
@@ -74,7 +74,7 @@
 		}
 
 		// We want to display a spinner until the authentication is loaded. This to avoid a glitch when either the landing page or effective content (sign-in / sign-out) is presented.
-		if ($authStore === undefined) {
+		if (isNullish($authStore)) {
```
Comment (peterpeterparker) [inline src/frontend/src/routes/+layout.svelte:77]: ditto

**Review by peterpeterparker — COMMENTED** [review]
Cool stuffs.

`[].find` or `[].filter` cannot return `null`. I'm fine using `isNullish` and `nonNullish` for those array operators but, if there is a way to improve the eslint rule, I have to say that I wont' be against neither - i.e. that would be actually nice.

**Comment by AntonioVentilii-DFINITY** [comment]
@peterpeterparker ah! I didn't think that some of the conditions were in place with a purpose...

so, either I put the @lint-exception in place there, or I can just discard this PR, what do you think? for example, if we want to keep the rule of this pr we would do:

```typescript
// eslint-disable-next-line nullish-check -- it is ok to display undefined becasue.....
if (value === null) {
```

**Comment by peterpeterparker** [comment]
> I didn't think that some of the conditions were in place with a purpose...

Yep. Checkout the definition of the store and how we handle errors in the services. On error we set `null`.

> so, either I put the @lint-exception

Lint-exception sounds good to me, I think it's a cool stuffs to have your eslint rule. 👍
We should just pay attention at the changes and not do a bulk change.

File: src/frontend/src/lib/components/ui/Json.svelte (hunk 3 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/ui/Json.svelte:25]: right! I am putting the eslint exception here

**Comment by AntonioVentilii-DFINITY** [comment]
Since the code changed too much since I last reviewed this PR, I will open a new one (https://github.com/dfinity/oisy-wallet/pull/2355) and close this one

**Pull request closed without merging**